ai-service/src/ai_service/db.py

```
@@ -8,39 +8,31 @@
-    Otherwise, create a new one.
-    """
-    repo_name = collection_name.split("__")[0]
-    for col in client.list_collections():
-        if col.name.startswith(f"{repo_name}__"):
-            return col
+def get_collection() -> chromadb.Collection:
+    """Get or create a ChromaDB collection using the current repo context."""
+    try:
+        repo_url = _current_repo_url.get()
+    except LookupError:
+        raise errors.DatabaseError.no_repo_context()
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)